src/config/rt.rs

```diff
@@ -281,6 +306,9 @@ impl RtcServe {
             proxy_ws: opts.proxy_ws,
             proxies,
             no_autoreload: opts.no_autoreload,
+            tls_private_key_path: opts.tls_private_key_path,
+            tls_public_key_path: opts.tls_public_key_path,
+            target_parent,
```

Comment:
Let's change this config interface a bit:
- Let's do all path parsing and RustlsConfig building here. The only thing which should come out of the runtime config now should be an `Option<RustlsConfig>`.
- This will ensure that the fallibility of this operation is isolated to the config phase, instead of waiting until the serve phase.
- As we do with our other relative path parsing, the relative path must be relative to the source where it is declared: Trunk.toml, index.html or CLI. If CLI, then it is relative to the CWD. As such, no `target_parent` is needed in this case.

Reply:
Thx for the swift and clear feedback.

> Let's do all path parsing and RustlsConfig building here. The only thing which should come out of the runtime config now should be an Option<RustlsConfig>.
> This will ensure that the fallibility of this operation is isolated to the config phase, instead of waiting until the serve phase.

Yes, that makes sense. It also immediately stops the command if the files cannot be found, and doesn't start the server.

> As we do with our other relative path parsing, the relative path must be relative to the source where it is declared: Trunk.toml, index.html or CLI. If CLI, then it is relative to the CWD. As such, no target_parent is needed in this case.

`relative to the source where it is declared: Trunk.toml, index.html or CLI` sounds reasonable. I also think that absolute paths currently don't work with my proposed implementation.

Hmm, ok. I already noticed `from_file` in `models.rs`, but now I finally understand what it's doing. It's patching/changing the paths of the relative paths defined in `Trunk.toml`, and converts them to absolute paths!

Which means I have to add two new blocks to `fn from_file` to overwrite the paths with the prefixed path (like target), and convert it to an absolute path. And then this means that I can remove `target_parent`, because in that case the path is relative to CWD, or it is an absolute path already.

So basically

```rs
if let Some(build) = cfg.build.as_mut() {
  // [...target]
  // [...dist]
  // TODO: add another two blocks for `tls_cert_path` and `tls_key_path`
}
```

Am I on the right track? Sorry, I'm a bit rusty, I haven't coded in Rust since a long time :)

Reply:
> I also think that absolute paths currently don't work with my proposed implementation.

Yea, we should only attempt to canonicalize a path after establishing that the given path does not exist.

> Am I on the right track?

That sounds correct! Happy hacking `:)`

Reply:
It's done. I think that the code now is better than it was before. Let me know if other parts should be improved